### DISCUSSION TITLE
Failed to resolve import "React"  when creating react npm package with webpack 5

### DISCUSSION THREAD

**Discussion by guitar9** [webpack/webpack#15142] [Q&A]
# Bug report

**What is the current behavior?**
I want to create a React Libary and push it to npm. With my webpack config I get the error: Failed to resolve import "React"

**If the current behavior is a bug, please provide the steps to reproduce.**
I have created for that a repo and asked already on stackoverflow
Stackoverflow: https://stackoverflow.com/questions/70651632/failed-to-resolve-import-react-in-npm-package-and-webpack-5
repo: https://github.com/guitar9/webpack-bug

**What is the expected behavior?**
The Library should take react and react-dom from parent. 

**Other relevant information:**
webpack version:  5.27.0
Node.js version: 17.3.0
Operating System: ubuntu
Additional tools:

**Comment by alexander-akait** [comment]
Because `react` is external

> The Library should take react and react-dom from parent.

Sorry no, it doesn't work like that, there is link https://dev.to/jimjunior/how-to-create-an-npm-library-from-react-components-2m2 in your questions, please use it for bundling library

**Reply by guitar9** [reply]
so the solution is not take another libary? This problem is so basic and there is no solution with webpack?

**Reply by guitar9** [reply]
I tried this out and with rollup I get this error: cannot set properties of undefined (setting 'runtime')
I created a new branch with name rollup at this repo
